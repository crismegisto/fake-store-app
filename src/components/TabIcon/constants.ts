import Routes from 'constants/routes';

interface Icons {
  [name: string]: string;
}

const ICONS: Icons = {
  [Routes.HomeNavigator]: 'home',
  [Routes.Support]: 'support-agent'
};

export default ICONS;
