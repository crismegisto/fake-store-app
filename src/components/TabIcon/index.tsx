import React from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';

import { lightGray, primary } from 'constants/colors';

import ICONS from './constants';

interface Props {
  name: string;
  focused: boolean;
}
function TabIcon({ name, focused }: Props) {
  return <Icon name={ICONS[name]} size={34} color={focused ? primary : lightGray} />;
}

export default TabIcon;
