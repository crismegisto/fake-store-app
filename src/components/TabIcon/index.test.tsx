import React from 'react';
import { screen, render } from '@testing-library/react-native';

import TabIcon from './index';

beforeEach(() => {
  render(<TabIcon name="home" focused={true} />);
});

describe('TabIcon snapshot', () => {
  test('match snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
