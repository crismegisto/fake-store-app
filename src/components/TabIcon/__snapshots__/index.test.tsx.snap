// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabIcon snapshot match snapshot 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": "#606c38",
        "fontSize": 34,
      },
      undefined,
      {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {},
    ]
  }
/>
`;
